### PR TITLE
asyncTimer

### PR DESCRIPTION
**Pull request by chrismatheson**
introduction of function which understands async return and instruments appropriately 

should resolve https://github.com/brightcove/hot-shots/issues/66

### CODE REVIEW

**Comment by bdeitte** [comment]
I see that the Node 8 says they failed, but I just reran it and it's fine.  I'll take an actual look at this PR (and your other one) for merging later tonight.  Thanks!

File: test/timer.js [hunk 1]
```diff
@@ -105,5 +105,32 @@ module.exports = function runTimerTestSuite() {
...
+
+      return instrumented().then(function() {
+
+        var stat = statsd.mockBuffer[0];
+        var name = stat.split(/:|\|/)[0];
+        var time = stat.split(/:|\|/)[1];
+
+        assert.equal(name, 'name-thingy');
+        assert.ok(parseFloat(time) >= 100);
+        assert.ok(parseFloat(time) < 110);
```
Comment (bdeitte) [inline test/timer.js]: I have the same worry here about creating a jittery test, perhaps a little higher?

**Comment by bdeitte** [comment]
One small comment.  Thank you for all of your work here, great to see!

Heads up @DavidWittman that there's another solution that came through here.

**Comment by chrismatheson** [comment]
sure thing @bdeitte 

I didn't want to step on any toes, but i was curious about the structure of the tests. Seems that the specific including of the test modules and wrapping in a top level describe is redundant if the recursive option for mocha is used.

I only noticed because my IDE wont recognise the mocha tests and let me easily run a single one etc.

Id be happy to refactor the tests a little if thats something that would be acceptable ?

**Comment by bdeitte** [comment]
@chrismatheson Always happy to take improvements.  :)  The tests were recently refactored to not be in one huge mess of a file, in https://github.com/brightcove/hot-shots/pull/79.  So it hasn't been around for long in the current form and if there's improvements to be made, I'll always review and pull in.

**Pull request merged**